extensions/s3/src/main/java/io/deephaven/extensions/s3/S3SeekableChannelProvider.java

```diff
@@ -62,48 +59,27 @@ final class S3SeekableChannelProvider implements SeekableChannelsProvider {
     private final S3AsyncClient s3AsyncClient;
     private final S3Instructions s3Instructions;
 
+    private final Map<URI, Long> uriToFileSize;
```

Comment:
`private final TObjectLongMap<URI> uriToFileSize;`
and
`        this.uriToFileSize = new TObjectLongHashMap<>();`

If we're sharing providers, we may want a pair class, and a `KeyedObjectHashMap`.

So maybe:
```
    private final SoftReference<KeyedObjectHashMap<URI, FileInfo>> uriToFileSize;

    private static final class FileInfo {
        private final URI uri;
        private final long size;
    }
```